LibreNMS/OS/Aix.php

```diff
@@ -36,9 +36,11 @@ public function discoverOS(Device $device): void
         $aix_descr = explode("\n", $device->sysDescr);
         // AIX standard snmp deamon
         if (! empty($aix_descr[1])) {
-            $device->serial = explode('Processor id: ', $aix_descr[1])[1];
+            $device->serial = snmp_get($this->getDeviceArray(), 'aixSeSerialNumber.0', '-Oqv', 'IBM-AIX-MIB');
```

Comment:
You should keep the original code as a backup in case the snmp_get does not return a value. Or the other way around, only use SNMP when the current code does not provide the value.

Reply:
I think, we already discussed this 2 year ago...
Original code takes wrong serial number. It is serial number of AIX itself probably. We need replace it with Hardware serial number. There is no reason to keep original code.

IBM-AIX-MIB has aixSeSerialNumber :
aixSeSerialNumber OBJECT-TYPE
        SYNTAX          DisplayString
        MAX-ACCESS      read-only
        STATUS          current
        DESCRIPTION
                "The serial number of the machine."
        ::= {aixAuxSystemEnvironment 7}

This is a standard for all AIX.
I am 100% sure that all versions of AIX provide a reply to this OID. And i tested this on AIX 5.3, 6.1, 7.1, 7.2 on production equipment.

Reply:
OK for this then.